x/upgrade/client/cli/tx.go

```diff
@@ -25,9 +24,19 @@ const (
 	FlagUpgradeInfo   = "info"
 )
 
+// GetTxCmd returns the transaction commands for this module
+func GetTxCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   types.ModuleName,
+		Short: "Upgrade transaction subcommands",
+	}
+	cmd.AddCommand(flags.PostCommands()...)
+
+	return cmd
+}
+
 // NewCmdSubmitUpgradeProposal implements a command handler for submitting a software upgrade proposal transaction.
 func NewCmdSubmitUpgradeProposal(clientCtx client.Context) *cobra.Command {
```

Comment:
This still needs to be updated: see https://github.com/cosmos/cosmos-sdk/pull/6639 for reference

Reply:
I actually went down this path, but it requires a bigger refactor:
- x/gov has a [`CLIHandlerFn`](https://github.com/cosmos/cosmos-sdk/blob/master/x/gov/client/proposal_handler.go#L18) that takes client.Context as arg
- x/distribution, x/params & x/upgrade use this handler. But PRs are still pending for using Protobuf in query: #6585 #6600 

what I propose:
- leave this PR scoped to x/upgrade, as initially intended
- I'll take care of one follow-up PR to refactor all x/{distribution,params,upgrade} at once once those PRs are merged